### ISSUE TITLE
Does the campaign price the measured map against AlphaQ's calibration-free allocation?

### ISSUE THREAD

**Issue by Alberto-Codes** [Alberto-Codes/vramfit#302]
## Question

AlphaQ allocates MoE expert bit-widths from the weight spectra alone, with
no calibration data and no measurement. **Does the #300 campaign price
vramfit's measured map against it, and does the record's "closest prior art"
move?**

## Why it is live

Chart #158's premise is that measured per-cell damage earns its cost against
heuristics. AlphaQ is the published method that attacks that premise most
directly, and no ticket carries it. A tracker search on 2026-08-16 for
`AlphaQ`, `spectral`, `heavy-tailed`, and `calibration-free` returned
nothing.

Raised 2026-08-16 while resolving #276, after that ticket's candidate list
misread the paper.

## What the paper states

[arXiv 2606.04980](https://arxiv.org/abs/2606.04980), ICLR 2026, Liquid AI.
Code at [Superone77/AlphaQ](https://github.com/Superone77/AlphaQ). From the
abstract:

> AlphaQ operationalizes this principle by measuring expert-wise spectral
> heavy-tailedness and solving a budget-constrained optimization problem
> that minimizes total quantization error under a global bit-budget
> constraint.

Its principle: experts with more heavy-tailed weight spectra are better
trained and take higher bit-widths. It reports near full-precision accuracy
at 3.5 bits average expert precision on Qwen1.5-MoE, and it beats
calibration-based baselines under matched budgets.

**It reads no calibration data and runs no forward pass.** vramfit measures
damage per cell on a rented H100 at $3.29/hr. That is the comparison.

## What the records already carry, and where they disagree

`docs/reference/prior-art.md:171-183` calls AlphaQ "the closest published
shape to chart #158's target" and records its baseline correctly.

Chart #158 Notes instead name `mlx_lm/quant/dynamic_quant.py` as the closest
prior art, found by #168. **The two records disagree and this ticket owns
the conflict.**

#276's candidate list read AlphaQ's baseline as its method, and my ruling
comment repeated the error. Both are
[corrected](https://github.com/Alberto-Codes/vramfit/issues/276#issuecomment-5310373201).

## The granularity problem

**AlphaQ's published granularity does not pack through llama.cpp.** It
allocates per expert, at 128 experts per MoE layer. #159 settled that
llama.cpp fuses each layer's experts into one tensor with one type, so the
recipe addresses 46 stacks and no flag reaches inside one. A per-expert width
needs a GGUF format change and a rewrite of every backend's `mul_mat_id`,
which chart #158 rules out of scope.

So a comparable arm pools alpha to the 46 stacks. **That is a modification of
the published method and any result must state it.** Whether pooled alpha
still carries the paper's signal is unmeasured, and this ticket does not
assume it does.

## Why the ordering is forced

**ADR-0027 requires every compared arm to run on one instrument against one
f16 base.** So an AlphaQ arm added after #300 runs does not join #300's
numbers. It re-runs the whole campaign.

Ruling this after the pod costs the campaign twice. Ruling it before costs
one comment.

## Candidates, none chosen here

- **No arm.** Record AlphaQ in prior art, state it as unpriced, and let the
  campaign run as #249 ruled. Cheapest. A reader may ask why the closest
  published competitor went unmeasured.
- **An AlphaQ arm at pooled alpha.** Compute alpha per expert matrix off the
  box's bf16 checkpoint, pool to 46 stacks, allocate at the ruled histogram
  of 35 at `Q4_0` and 11 at `Q2_0`. Prices measurement against a
  calibration-free prior on one instrument.
- **Alpha replaces arm 4.** #300's class-wise arm needs a 12-of-23 tie-break
  that no record rules, and it tests a hand-made tensor-class prior. Alpha is
  map-free too, and it is published. This swaps a weak opponent for the real
  one, and it retires the tie-break question.
- **Alpha replaces the blind arm.** Refused on its face and listed for
  completeness. #276 ruled the blind arm uninformative by construction, and
  alpha is informative by design.

## Cost, and what is unknown

Computing alpha needs a spectral fit per expert matrix over 5,888 expert
tensors. It reads no calibration data and runs no forward pass, and the box
already holds the 61.32 GiB bf16 checkpoint (#266). So it is box-side and
buys no pod time.

**The compute cost is unmeasured and this ticket does not estimate it.** A
spike sizes it before any arm is bought. The pack and evaluation cost is
known: about 40 s of quantize and 80 s of evaluation per arm, on #229's
timings.

## Also unread

[BitsMoE](https://arxiv.org/pdf/2606.00079), spectral energy-guided bit
allocation for MoE, appears to allocate per expert and per projection. **A
first pass could not confirm its method or whether it needs calibration
data.** It is recorded here as unread rather than cited. If a projection-level
published prior exists, it reaches #300's arm 4 directly.

## What closing looks like

A dated maintainer ruling in chart #158's Notes, and any `prior-art.md`
change. This ticket stores no decision.

## Blocks

#300. Wired as an edge, on the ADR-0027 argument above. Cut the edge if the
ruling is "no arm".

**Comment by Alberto-Codes** [comment]
## Spike findings, 2026-08-16

The spike ran box-side. It bought no pod and it spent no money. This
comment stores no decision.

## The method, read from the paper and the code

[arXiv 2606.04980](https://arxiv.org/abs/2606.04980) is "AlphaQ:
Calibration-Free Bit Allocation for Mixture-of-Experts Quantization",
by Yang, Ma, Conzelmann, Zheng, Mahoney, Rusch, and Liu. The abstract
confirms the ticket's reading. AlphaQ measures expert-wise spectral
heavy-tailedness and solves a budget-constrained optimization problem.
It reads no calibration data and it runs no forward pass.

[Superone77/AlphaQ](https://github.com/Superone77/AlphaQ) states the
mechanics the paper omits.

- `utils_alpha.py` computes PL_Alpha_Hill per `nn.Linear` weight. FARMS
  is on by default. It samples at most 256 sub-blocks of 128 by 128, at
  stride 128. It takes the squared singular values of every block, sorts
  them, and applies a Hill estimator at `k_frac` 0.1.
- `build_and_solve_global_milp` allocates. Its per-cell cost is
  `(alpha0 / alpha)**gamma * var(W) * 2**(-2b)`, where `alpha0` is the
  median alpha. It minimizes the total under one average bit-width
  constraint.
- **The shipped objective carries a weight-variance term beside the
  spectral term.** The abstract names the spectral term alone. I read
  the abstract and the code, not the full PDF, so the paper's body may
  derive the variance term. This matters below.
- The repository documents `gamma` 1.0 and its worked example uses 1.0.
  The recipe it ships under `docs/bit_recipes_2layer` is named
  `gamma10.0`. This spike reports both.

`docs/alpha_qwen3_2layer.csv` shows AlphaQ scoring each expert **per
projection**, not per expert. Its shipped recipe assigns a width to each
`experts.down_proj.expert_<n>`.

## Cost, which #302 states is unmeasured

The 30B bf16 checkpoint is resident, at 62 GiB in `~/models`. The count
matches the record: 23 MoE layers, 128 experts, 2 projections, 5,888
expert matrices. Rows are 2688 and 1856, and every stack holds
638,582,784 parameters.

| figure | measured |
|--------|----------|
| wall-clock, all 5,888 matrices | **572.4 s** |
| peak VRAM | **0 B** |
| peak host memory | 2.76 GiB |
| checkpoint read | 34.1 s of the 572.4 s |

**The method needs no GPU.** It ran on the box's 8 CPU threads. A first
attempt ran the SVDs on the 4090 and measured 5 times slower, because
cuSOLVER batches 128 by 128 decompositions badly. So alpha costs about
10 minutes of one CPU and no pod time.

The implementation reproduces `utils_alpha.py` within 3e-6 relative alpha
on spot checks. It pins `FARMS_SEED` to 0, which the reference leaves
unseeded. Two of 5,888 matrices returned a non-finite alpha through a
`eigvalsh` convergence failure. Both were recomputed through the
reference path and both are finite.

## The pooling rule, and why

**This is a modification of the published method.** AlphaQ allocates per
expert. #159 settled that llama.cpp serves 46 stacks.

The rule: **sum each stack's 128 per-expert AlphaQ cost terms, then
allocate over the 46 stacks.** A stack takes one width, so its
contribution to AlphaQ's own objective is the sum over its experts. That
is AlphaQ's MILP with the feasible set cut to what the runtime serves,
and it changes no coefficient. Every expert matrix holds equal
parameters, so the sum is proportional to the mean.

A second rule, pooling alpha and variance by mean, returns the **same 35
stacks**. So the result does not turn on the pooling choice.

**Pooling keeps most of the signal.** Alpha's spread across the 5,888
matrices sits 80.41 % between the 46 stacks and 19.59 % within them. The
46 stack means carry a standard deviation of 0.5146, against a median
within-stack figure of 0.2435. So the cross-expert axis AlphaQ exploits
holds a fifth of alpha's variation on this target.

I ran AlphaQ's `build_and_solve_global_milp` unmodified over the 46
stacks, at a budget of 3.5217 nominal bits. It returned the 35/11
histogram. **It returned a suboptimal set, by one stack.** The objective
magnitude is 2.6e-4, which falls under CBC's absolute tolerance. Scaling
the objective by 1e6 returns the true optimum, which equals the ranking
below. This is an observation about the reference code and it does not
move the finding.

## The allocation at the ruled histogram

35 stacks at `Q4_0` and 11 at `Q2_0`, dense at nominal 8. That is 3.9620
bits per expert weight, or 3.5217 nominal. AlphaQ's headline operating
point is 3.5 bits average expert precision, so the budgets nearly
coincide.

At `gamma` 1.0 the 11 stacks held at `Q2_0` are:

```
blk.1.ffn_up_exps     blk.1.ffn_down_exps
blk.3.ffn_up_exps     blk.3.ffn_down_exps
blk.6.ffn_down_exps   blk.8.ffn_down_exps
blk.10.ffn_down_exps  blk.22.ffn_down_exps
blk.27.ffn_down_exps  blk.29.ffn_down_exps
blk.47.ffn_down_exps
```

It relieves 21 of 23 `ffn_up_exps` and 14 of 23 `ffn_down_exps`. It
splits 7 layers, which a layer-keyed arm cannot express.

## The overlap, which is the number #302 asks for

Arm 1's top 17 MoE layers are the same set under both readings of the
sensitivity map, by damage at nominal 2 and by the nominal 2 minus
nominal 4 gain. So arm 1's 35 is 34 paired stacks plus one stack of
layer 29 or 31, which the record leaves open.

| comparison | overlap |
|------------|---------|
| **AlphaQ against arm 1** | **25 or 26 of 35** |
| a uniform random 35-of-46 draw against arm 1 | 26.63 of 35 |
| AlphaQ against arm 4, `ffn_up` tie-broken by layer index | 24 of 35 |

**AlphaQ's pooled pick agrees with the measured map no more than chance
does.** At `gamma` 2.0 and above the overlap reaches 27 of the 34 paired
stacks, which is still at chance.

## What drives that, and it is not the spectral term

Relief prefers a low alpha and a high variance. Against measured damage
at nominal 2:

| term | `ffn_up_exps` | `ffn_down_exps` | direction |
|------|---------------|-----------------|-----------|
| mean alpha | -0.562 | -0.488 | **agrees** with the map |
| mean variance | -0.808 | -0.742 | **opposes** the map |
| published objective | -0.636 | -0.328 | opposes the map |

Spearman rho over the 23 MoE layers. **AlphaQ's spectral term leans the
same way as vramfit's measurement. Its variance term leans harder the
other way, and the product follows the variance.** Weight variance rises
steeply with depth on this checkpoint, at rho +0.950 and +0.867.

The consequence is sharp. Layers 1 and 3 carry the highest measured
damage at nominal 2, at 0.38097 and 0.35843. **The published objective
sends all four of their stacks to `Q2_0`**, because their weight
variance is the lowest in the model.

An alpha-only ranking, dropping the variance term, reaches 28 of 35
against arm 1 and relieves all 23 `ffn_up_exps`. It still holds
`blk.1.ffn_down_exps` and `blk.3.ffn_down_exps` at `Q2_0`.

## The options, restated against the data

- **No arm.** The cost argument for an arm improved. Alpha is 10 minutes
  of CPU and $0, against the $0.11 of pack and evaluation an arm adds to
  the booked pod.
- **An AlphaQ arm at pooled alpha.** The overlap says it buys a genuinely
  different recipe. It does not say the recipe is good. The arm would
  price a published calibration-free allocation that disagrees with the
  measurement on the two highest-damage layers.
- **Alpha replaces arm 4.** The data refuses this on its stated
  rationale. Arm 4 relieves 23 `ffn_down_exps` and 12 `ffn_up_exps`.
  AlphaQ relieves 14 and 21. They are near mirror images, and they
  overlap at 24 of 35, under chance. Alpha is not a stronger version of
  the tensor-class prior. It is a different prior that points the other
  way.
- **Alpha replaces the blind arm.** Still refused on its face. The
  chance-level overlap does not make alpha uninformative. Alpha is
  deterministic and weight-derived, where #276's arm is uninformative by
  construction.

**A fork this spike opened.** An arm at the published objective mostly
prices the variance term. An arm at alpha only prices the HT-SR claim
the paper argues. The second is a second modification stacked on the
pooling. Any arm states which objective it ran.

## Caveats

- AlphaQ reports on Qwen1.5-MoE, Mixtral-8x7B, and DeepSeekV2-Lite. Those
  carry gated experts at three matrices. This target's experts are
  ungated at two.
- The `2**(-2b)` cost is a uniform-quantizer proxy. `Q4_0` and `Q2_0`
  carry per-block scales at 32 and 64 weights, so the variance term
  transfers only in part.
- The ruled 35/11 histogram fixes the count, which #249 requires for a
  matched file size. That turns AlphaQ's MILP into a ranking.
- The overlap compares recipes and not damage. Only the campaign prices
  which recipe serves better.

## BitsMoE, recorded as read

[arXiv 2606.00079](https://arxiv.org/abs/2606.00079) is "BitsMoE:
Efficient Spectral Energy-Guided Bit Allocation for MoE LLM
Quantization", submitted 2026-05-22. It decomposes each MoE layer by SVD
into a shared basis and expert-specific spectral factors. It allocates
per spectral factor and solves an integer linear program.

**BitsMoE needs calibration data.** Its abstract states an
"activation-aware reconstruction surrogate". So it is not a
calibration-free prior and it does not attack this chart's premise.
Its granularity is finer than per expert, so it reaches neither
llama.cpp's 46 stacks nor #300's arm 4. **No follow-up ticket is
needed, and #302's "Also unread" item closes here.**

## Over to the maintainer

#302 is `chart:discuss` and it resolves through your ruling. Nothing
lands in the chart's `## Decisions so far` until then. #300 stays
blocked.

**Comment by Alberto-Codes** [comment]
## Resolved, 2026-08-16

**Maintainer ruling: the campaign buys no AlphaQ arm.** The record is chart
#158's Notes, at the entry dated 2026-08-16 from #302. This ticket stores no
decision.

The ruling also refuses the arm-4 substitution this ticket listed. AlphaQ
and arm 4 overlap at 24 of 35, which is under chance, and they relieve
opposite projections.

Records:

- [chart #158 Notes](https://github.com/Alberto-Codes/vramfit/issues/158)
- [spike findings](https://github.com/Alberto-Codes/vramfit/issues/302#issuecomment-5310542313)

## What this ticket changed

The conflict it owned is settled. `docs/reference/prior-art.md` keeps
AlphaQ as the closest published shape, and it now also stands as unpriced
with a measured reason. The chart's `mlx_lm/quant/dynamic_quant.py` entry
stays, because the two records describe different things. One names the
closest shipping code and the other names the closest published method.

**A `prior-art.md` change follows in this session.** It records AlphaQ as
unpriced and carries the spike's numbers.

## Deferrals landed

- **BitsMoE closes here.** arXiv 2606.00079 needs calibration data and
  allocates finer than per expert. It reaches neither the 46 stacks nor
  #300's arm 4. No follow-up ticket exists, and chart Notes carry the fact.
- **An objective fork stays open and unbought.** AlphaQ's shipped objective
  prices its variance term more than its spectral term. An alpha-only arm
  would price the HT-SR claim instead. The ruling buys no arm, so neither
  runs. Chart Notes carry the measurement that would justify revisiting.

## Rule deviation, flagged not hidden

`.claude/rules/charting.md` caps a session at one resolved decision ticket.
This session recorded three maintainer rulings: #302's, plus arm 4's
tie-break and the six-arm count on #300. The two extra rulings own no
ticket of their own. Both were unrouted conflicts that #276 left behind,
and the maintainer gave all three in one exchange. Raising it here rather
than deviating in silence.

## Unblocks

#300. The dependency edge is cut.
